src/components/loader-spinner/loader-spinner.spec.tsx

```diff
@@ -0,0 +1,290 @@
+      expect(visibleLabelElement).toBeVisible();
+    });
+  });
+
+  describe("when custom props are passed", () => {
+    it.each(sizes)(
+      "when the 'size' prop is passed as `%s` the wrapper has the correct dimensions",
+      (spinnerSizes) => {
+        render(<LoaderSpinner size={spinnerSizes} />);
+        const wrapperElement = screen.getByRole("status");
+
+        expect(wrapperElement).toHaveStyle(
+          `width: ${LOADER_SPINNER_SIZE_PARAMS[spinnerSizes].wrapperDimensions}px`
+        );
+        expect(wrapperElement).toHaveStyle(
+          `height: ${LOADER_SPINNER_SIZE_PARAMS[spinnerSizes].wrapperDimensions}px`
+        );
+      }
+    );
+
+    it.each(sizes)(
+      "when the 'size' prop is passed as `%s` the wrapper has the correct minimum dimensions",
+      (spinnerSizes) => {
+        render(<LoaderSpinner size={spinnerSizes} />);
+        const wrapperElement = screen.getByRole("status");
+
+        expect(wrapperElement).toHaveStyle(
+          `min-width: ${LOADER_SPINNER_SIZE_PARAMS[spinnerSizes].wrapperDimensions}px`
+        );
+        expect(wrapperElement).toHaveStyle(
+          `min-height: ${LOADER_SPINNER_SIZE_PARAMS[spinnerSizes].wrapperDimensions}px`
+        );
+      }
+    );
+
+    it.each(sizes)(
+      "when the 'size' prop is passed as `%s` the inner arc has the correct stroke width",
+      (spinnerSizes) => {
+        render(<LoaderSpinner size={spinnerSizes} />);
+        const innerArcSvgElement = screen.getByTestId("inner-arc");
+
+        expect(innerArcSvgElement).toHaveStyle(
+          `stroke-width: ${LOADER_SPINNER_SIZE_PARAMS[spinnerSizes].strokeWidth}px`
+        );
+      }
+    );
+
+    it.each(sizes)(
+      "when the 'size' prop is passed as `%s` the outer arc has the correct stroke width",
+      (spinnerSizes) => {
+        render(<LoaderSpinner size={spinnerSizes} />);
+        const outerArcSvgElement = screen.getByTestId("outer-arc");
+
+        expect(outerArcSvgElement).toHaveStyle(
+          `stroke-width: ${LOADER_SPINNER_SIZE_PARAMS[spinnerSizes].strokeWidth}px`
+        );
+      }
+    );
+
+    it("when 'showSpinnerLabel' is `true` the visible label is rendered in the live region container", () => {
+      render(<LoaderSpinner showSpinnerLabel />);
+      const wrapperElement = screen.getByRole("status");
+      const visibleLabelElement = screen.getByTestId("visible-label");
+
+      expect(wrapperElement).toHaveTextContent("Loading...");
+      expect(visibleLabelElement).toBeVisible();
+    });
+
+    it("when 'showSpinnerLabel' is `false` the visible label is not rendered in the live region container", () => {
+      render(<LoaderSpinner showSpinnerLabel={false} />);
+      const wrapperElement = screen.getByRole("status");
+      const visibleLabelElement = screen.queryByTestId("visible-label");
+
+      expect(wrapperElement).toHaveTextContent("Loading...");
+      expect(visibleLabelElement).not.toBeInTheDocument();
+    });
```

Comment:
**comment**: thanks for amendment here @tomdavies73 👍🏼 

I just wanted to suggest an alternative approach for us to explore in future PRs. Instead of using a test-id approach, we could assert that the label inside the live region is present in the document but visually hidden. This ensures the intended behaviour is directly tested while making the test resistant to markup changes. What do you think?

```tsx
import { render, screen, within } from "@testing-library/react";

it("when 'showSpinnerLabel' is `false`, loading label has visually hidden styling", () => {
  render(<LoaderSpinner showSpinnerLabel={false} />);
  const wrapperElement = screen.getByRole("status");
  const label = within(wrapperElement).getByText("Loading..."); // query throws error if label is not in the document
  
  expect(label).toHaveStyle({
    border: "0",
    height: "1px",
    margin: "-1px",
    overflow: "hidden",
    padding: "0",
    position: "absolute",
    width: "1px",
    "white-space": "nowrap",
  });
});
```

Reply:
Yeah looks good, I don't mind using this approach here either but happy to keep the current approach too